src/Instrumentation/IO/_register.php

```diff
@@ -2,6 +2,10 @@
 
 declare(strict_types=1);
 
-assert(extension_loaded('otel_instrumentation'));
+use OpenTelemetry\Contrib\Instrumentation\IO\IOInstrumentation;
 
-\OpenTelemetry\Contrib\Instrumentation\IO\IOInstrumentation::register();
+if (extension_loaded('otel_instrumentation') === true) {
```

Comment:
We could also reverse condition to check it first and get rid of else branch
```
if (extension_loaded('otel_instrumentation') === false) {
   trigger_error('The otel_instrumentation extension must be loaded in order to autoload the OpenTelemetry IO 
   auto-instrumentation', E_USER_WARNING);
}
IOInstrumentation::register();
```

Overall LGTM

Reply:
> This will _always_ call the `IOInstrumentation::register()` static methods, which we don't want to do when the `otel_instrumentation` extension is not available/loaded. However, we could do this approach if we add a `return;` statement just after the `trigger_error`. Is this preferred over the current code?

Yes, I forgot to add return. Don't think that there is a convention. This is just my preference.

Reply:
I have applied your feedback 👍